cli/azd/pkg/apphost/generate.go

```diff
@@ -431,6 +431,13 @@ func GenerateProjectArtifacts(
 	manifest *Manifest,
 	appHostProject string,
 ) (map[string]ContentsAndMode, error) {
+	// panic if projectName contains spaces. This is because the project name is used to generate the name of containers
+	// and other resources, and spaces are not allowed in those names.
+	if strings.Contains(projectName, " ") {
```

Comment:
Just wondering, how would we maintain this code over time?

There are many, many other disallowed characters. `apphost.generate` knows not a lot about project name validation.

I'm also fine if we want to track the removal with the linked naming issue #4400. I would lean towards just removing this -- we can catch this via code review right now.